### ISSUE TITLE
Octasine Ver. 0.8.2 crashes on launch.

### ISSUE THREAD

**Issue by BubbaByce** [greatest-ape/OctaSine#135]
**Describe the bug**
The bug is my DAW crashing when I try to launch Octasine.

**What did you try to do**
1. Find Octasine in the plugin directory.
2. Drag Octasine to the plugin import spot.
3. DAW Crashes.

**Expected behavior**
I expected the program to run when I launched it in my DAW.

**Screenshots**
https://user-images.githubusercontent.com/121590713/209895596-e7d4c89e-3ec6-4a1d-a7f0-2dd5e351473c.png 
https://user-images.githubusercontent.com/121590713/209895601-6ef73c08-3b06-49d3-8cd8-27cd4a9b78fc.png 

**Software (please complete the following information):**
 - OctaSine version 0.8.2. 
 - DAW [My DAW is Waveform. I have also tried Soundbridge and that also crashes.]
 - OS: [Windows 10]

**Additional context**
I put "OctaSine v0.8.dll" in C:\Program Files\VstPlugins

**Log file contents**
Please include the contents of the OctaSine log file (`OctaSine.log`). Its location depends on your operating system (replace __Alice__ with your username):
- _On macOS_: /Users/__Alice__/Library/Application Support/com.OctaSine.OctaSine/OctaSine.log
- _On Windows_: C:\\Users\\__Alice__\\Documents\\OctaSine\\OctaSine.log
- _On Linux_: /home/__Alice__/.config/octasine/OctaSine.log

```
Octasine.log:
20:15:01 [INFO] init

20:15:01 [INFO] OS: Windows 10.0.19045 (Windows 10 Home) [64-bit]
20:15:01 [INFO] OctaSine build: v0.8.2 (6abeae3) (gl)
20:15:01 [ERROR] thread '<unnamed>' panicked at 'ERROR: 0:13: 'location' : not supported for this version or the enabled extensions 
ERROR: 0:13: '' : missing #endif 
ERROR: 0:13: '' : compilation terminated 
ERROR: 3 compilation errors.  No code generated.

': C:\Users\runneradmin\.cargo\registry\src\git

Soundbridge.log (I think it might be helpful):
[SoundBridge.log](https://github.com/greatest-ape/OctaSine/files/10317304/SoundBridge.log)
hub.com-1ecc6299db9ec823\iced_glow-0.5.1\src\program.rs:77

**Comment by BubbaByce** [comment]
I would also like to add that 0.7 works fine.
https://user-images.githubusercontent.com/121590713/209916465-2dac7a4a-ea1a-46f7-90da-a624e28299b3.png

**Comment by greatest-ape** [comment]
Thanks!

**Comment by greatest-ape** [comment]
Could you please try this build?

[OctaSine-tmp-2023-01-01-Windows.zip](https://github.com/greatest-ape/OctaSine/files/10329327/OctaSine-tmp-2023-01-01-Windows.zip)
[OctaSine-tmp-2023-01-01-Ubuntu-20_04.zip](https://github.com/greatest-ape/OctaSine/files/10329328/OctaSine-tmp-2023-01-01-Ubuntu-20_04.zip)
[OctaSine-tmp-2023-01-01-macOS.zip](https://github.com/greatest-ape/OctaSine/files/10329329/OctaSine-tmp-2023-01-01-macOS.zip)

**Comment by ecashin** [comment]
I'm not sure this is the same, but if I create an empty Renoise session and attempt to load a freshly compiled OctaSine 0.8.2 VST2 on Linux, Renoise crashes with this log output.

It looks like `fatal runtime error: Rust panics must be rethrown` is maybe coming from OctaSine itself.

```
Renoise LOG> Application: Enter MainLoop...                                                                                                          
Renoise LOG> VstPlugs: Trying to instantiate /home/ecashin/.vst/OctaSine-0.8.2.so                                                                    
Renoise LOG> VstPlugs: Analyzing VST plugin 'Joakim Frostegard: OctaSine-0.8.2', Unique ID:1438048625, Version:2400                                  
Renoise LOG> VstPlugs: Instantiate OK                                                                                                                
Renoise LOG> VstWindow: Creating a new X11Window...                                                                                                  
Renoise LOG> VstWindow: Setting X11Window properties...
Renoise LOG> VstWindow: Give the plug access to the X11Window...
fatal runtime error: Rust panics must be rethrown
Renoise LOG> CrashLog: Caught signal: SIGABRT (6) ...
Renoise LOG> VstPlugs: Caught exception in VSTPlug::Dispatch
Renoise LOG> CrashLog: Caught signal: SIGSEGV (11) ...
Renoise LOG> VstPlugs: Instantiate FAILED (caught structured exception) !!!
Renoise LOG> CrashLog: Caught signal: SIGSEGV (11) ...
Renoise LOG> CrashLog: /lib/x86_64-linux-gnu/libpthread.so.0(+0x12980) [0x7ff6393e7980]
Renoise LOG> CrashLog: /home/ecashin/opt/bin/renoise() [0xb30cd4]
Renoise LOG> CrashLog: [(nil)]
Renoise LOG> CrashLog: [0x7ffee38b9c40]
Renoise LOG> Application: Caught an unhandled fatal exception (Thread: GUI)!
Renoise LOG> Application: Saving a backup...
Renoise LOG> Error Message: A fatal error or crash occurred (unhandled exception in thread: GUI).
Renoise LOG> Error Message: A backup of the current document was saved at
Renoise LOG> Error Message: '/home/ecashin/.config/Renoise/V3.4.2/CrashBackups/Untitled_1_12_2023__12_40_crash.xrns'...
Renoise LOG> Error Message: This either happened because of a bug in Renoise, or because of a bug in one of its loaded components (plugins). Please c
ontact <bug@renoise.com> and report this problem, so that it can be fixed.
Renoise LOG> Error Message: Note: It's very important that we know exactly what has happened (what you were doing before this message popped up), or
the problem cannot be replicated/analyzed. Please include a description of what you were doing and which components were being used...
Renoise LOG> Application: Terminating...
```

**Comment by greatest-ape** [comment]
> I'm not sure this is the same, but if I create an empty Renoise session and attempt to load a freshly compiled OctaSine 0.8.2 VST2 on Linux, Renoise crashes with this log output.
> 
> It looks like `fatal runtime error: Rust panics must be rethrown` is maybe coming from OctaSine itself.
> 
> (I couldn't try the binaries like the "tmp" build above because of a library incompatibility on my system. Building from source works.)
> 
> ```
> Renoise LOG> Application: Enter MainLoop...                                                                                                          
> Renoise LOG> VstPlugs: Trying to instantiate /home/ecashin/.vst/OctaSine-0.8.2.so                                                                    
> Renoise LOG> VstPlugs: Analyzing VST plugin 'Joakim Frostegard: OctaSine-0.8.2', Unique ID:1438048625, Version:2400                                  
> Renoise LOG> VstPlugs: Instantiate OK                                                                                                                
> Renoise LOG> VstWindow: Creating a new X11Window...                                                                                                  
> Renoise LOG> VstWindow: Setting X11Window properties...
> Renoise LOG> VstWindow: Give the plug access to the X11Window...
> fatal runtime error: Rust panics must be rethrown
> Renoise LOG> CrashLog: Caught signal: SIGABRT (6) ...
> Renoise LOG> VstPlugs: Caught exception in VSTPlug::Dispatch
> Renoise LOG> CrashLog: Caught signal: SIGSEGV (11) ...
> Renoise LOG> VstPlugs: Instantiate FAILED (caught structured exception) !!!
> Renoise LOG> CrashLog: Caught signal: SIGSEGV (11) ...
> Renoise LOG> CrashLog: /lib/x86_64-linux-gnu/libpthread.so.0(+0x12980) [0x7ff6393e7980]
> Renoise LOG> CrashLog: /home/ecashin/opt/bin/renoise() [0xb30cd4]
> Renoise LOG> CrashLog: [(nil)]
> Renoise LOG> CrashLog: [0x7ffee38b9c40]
> Renoise LOG> Application: Caught an unhandled fatal exception (Thread: GUI)!
> Renoise LOG> Application: Saving a backup...
> Renoise LOG> Error Message: A fatal error or crash occurred (unhandled exception in thread: GUI).
> Renoise LOG> Error Message: A backup of the current document was saved at
> Renoise LOG> Error Message: '/home/ecashin/.config/Renoise/V3.4.2/CrashBackups/Untitled_1_12_2023__12_40_crash.xrns'...
> Renoise LOG> Error Message: This either happened because of a bug in Renoise, or because of a bug in one of its loaded components (plugins). Please c
> ontact <bug@renoise.com> and report this problem, so that it can be fixed.
> Renoise LOG> Error Message: Note: It's very important that we know exactly what has happened (what you were doing before this message popped up), or
> the problem cannot be replicated/analyzed. Please include a description of what you were doing and which components were being used...
> Renoise LOG> Application: Terminating...
> ```

Thanks! It is likely the same issue. Could you please try building from https://github.com/greatest-ape/OctaSine/tree/work-2023-01-01? It is the branch with the possible fix.

**Comment by ecashin** [comment]
> Thanks! It is likely the same issue. Could you please try building from https://github.com/greatest-ape/OctaSine/tree/work-2023-01-01? It is the branch with the possible fix.

Yes, I tried during lunch but encountered a missing dependency.

```
bash$ cargo build --release -p octasine-vst2-plugin
    Updating git repository `https://github.com/bungoboingo/iced.git`
error: failed to load source for dependency `iced_core`

Caused by:
  Unable to update https://github.com/bungoboingo/iced.git?branch=feat/background-gradients#7c16569f

Caused by:
  object not found - no match for id (7c16569f42651240a25120c6a4f094eb80c264a1); class=Odb (9); code=NotFound (-3)
```

**Comment by ecashin** [comment]
I tried moving `Cargo.lock` out of the way, but without the pinning, there's a compile error because of an incompatibility.

**Comment by greatest-ape** [comment]
@ecashin ok, thanks. I think that’s because of force pushes in the upstream fix branch. I’ll se if I can fix them.

Is the other issue still present in 0.8.1? In that case, yes, please open a new issue!

**Comment by ecashin** [comment]
> @ecashin ok, thanks. I think that’s because of force pushes in the upstream fix branch. I’ll se if I can fix them.
> 
> Is the other issue still present in v0.8.1? In that case, yes, please open a new issue!

Yes, it is, so I created #136 to track the problem.  Thanks much!

**Comment by greatest-ape** [comment]
@BubbaByce Could you please try out this build?

[OctaSine-tmp-2023-02-12-Windows.zip](https://github.com/greatest-ape/OctaSine/files/10717136/OctaSine-tmp-2023-02-12-Windows.zip)

**Comment by greatest-ape** [comment]
This is most likely fixed in the latest release, 0.8.3. If not, please reopen the issue.
